combat-trainer.lic

```diff
@@ -1131,22 +1131,43 @@ class SpellProcess
   private
 
   def check_slivers(game_state)
+    return unless DRStats.moon_mage?
     return if game_state.casting
     return if @tk_ammo
     return unless @tk_spell
     return if DRSpells.slivers
     return if UserVars.moons['visible'].empty?
-
-    fput('prep moonblade')
-    if DRSkill.getrank('Lunar Magic') < 300
-      pause 3
-    elsif DRSkill.getrank('Lunar Magic') < 400
-      pause 2
-    else
-      pause
+    # With low skill you may fail to create the slivers
+    # when breaking your moonblade. If so, keep retrying.
+    retry_count = 0
+    max_retries = 3
+    created_slivers = false
+    # We don't need to cast the spell a fancy way
+    # so rather than asking users to define a new waggle
+    # then simply cast the spell at minimum prep.
+    moonblade_spell = get_data('spells').spell_data['Moonblade'].dup
+    # Minimally prepare the spell so we can cast it quickly.
+    moonblade_spell['mana'] = 1
+    # Try, try, try to break moonblade into slivers.
+    # For novices, this may fail a couple times.
+    while !created_slivers && retry_count < max_retries
+      # For skilled mages, try to snap cast
+      # unless we failed the first time.
+      if retry_count == 0
+        case DRSkill.getrank('Lunar Magic')
+        when 200..299
+          moonblade_spell['prep_time'] = 3
+        when 300..399
+          moonblade_spell['prep_time'] = 2
+        when 400..Float::INFINITY
+          moonblade_spell['prep_time'] = 1
+        end
+      end
+      DRCA.cast_spell(moonblade_spell, @settings)
+      created_slivers = bput("break moonblade", "The slivers drift about", "dissipate without any benefit", "Break what?") == "The slivers drift about"
+      retry_count = retry_count + 1
     end
-    fput("cast #{UserVars.moons['visible'].first}")
-    fput('break moonblade')
+    DRC.message("Failed to create slivers for telekinetic ammo after #{retry_count} attempts") unless created_slivers
   end
```

Comment:
Not all mages can snap cast the moonblade spell nor every attempt to break a summoned moonblade may work when trying to create slivers as telekinetic ammo. This change adds logic to retry casting the spell up to 3 times. Maintains the original "snap cast" prep time logic that was here before.